src/main/resources/public/modules/viescolaire/directives/grouping/groupingItem/grouping-item.ts

```diff
@@ -0,0 +1,151 @@
+import {idiom, ng} from "entcore";
+import {Grouping, GroupingClass} from "../../../models/common/Grouping";
+import {ILocationService, IScope, IWindowService} from "angular";
+import {GroupingService} from "../../../services/GroupingService";
+import {Structure} from "../../../models/personnel/Structure";
+import {Classe} from "../../../models/personnel/Classe";
+
+
+interface IViewModel {
+    $onInit(): any;
+
+    $onDestroy(): any;
+
+    updateGroupingItem(grouping: Grouping, name: string): void;
+
+    deleteGroupingItem(grouping: Grouping): void;
+
+    addGroupingAudience(grouping: Grouping, classOrGroup: Classe): void;
+
+    deleteGroupingAudience(grouping: Grouping, classOrGroup: Classe): void;
+
+    getAllClass(): Classe[];
+
+    deleteClassItem(groupingClass: GroupingClass, grouping: Grouping, classet: Classe);
+
+    groupingItem: Grouping;
+
+    structure: Structure;
+
+    groupingClassItem: GroupingClass;
+
+    groupingClass: GroupingClass[];
+
+    lang: typeof idiom;
+
+}
+
+class Controller implements ng.IController, IViewModel {
+    $parent: any;
+    viewModel: IViewModel;
+    groupingItem: Grouping;
+    structure: Structure;
+    onDeleteGroupingItem: any;
+    onUpdateGroupingItem: any;
+    onAddGroupingAudienceItem: any;
+    onDeleteGroupingAudienceItem: any;
+    groupingClassItem: GroupingClass;
+    groupingClass: GroupingClass[];
+    classe: Classe;
+    lang: typeof idiom;
+
+    constructor(private $scope: IScope, private $location: ILocationService, private $window: IWindowService, private groupingService: GroupingService) {
+        this.$scope['vm'] = this;
+        this.lang = idiom;
+    }
+
+    $onInit = async () => {
+        this.groupingClass.forEach(grouping => {
+            this.groupingClassItem = this.groupingClass.find(groupingClass => groupingClass.grouping.id == this.groupingItem.id);
+        })
+        this.groupingItem.class.forEach(classeItem => {
+            this.classe = this.structure.classes.find(classe => classe.id == classeItem.id);
+            this.groupingClassItem.classes.push(this.classe);
+        })
```

Comment:
tu as vraiment besoin que "classe" soit ne variable global ?

this.groupingClassItem.classes = this.structure.classes.filter(classe => this.groupingItem.class.find(classeItem => classeItem.id === classe.id));
pas mieux ça ?